src/zircon/apis/frontend.go

```diff
@@ -0,0 +1,57 @@
+package apis
+
+import (
+	"crypto/sha256"
+	"fmt"
+)
+
+// A hash of a write at a particular offset with a particular length and data.
+type CommitHash string
+// An ID of an access control entity. If positive, represents a named entity. If negative, represents a defined ACL.
+// If zero, means "everyone has access".
+type AccessControlID int64
+
+// Means "everyone has access"
+const UnrestrictedAccess AccessControlID = 0
+
+type Frontend interface {
+	// Reads metadata about a particular chunk.
+	ReadMetadata(chunk ChunkNum) (AccessControlID, Version, ChunkReference, []ServerAddress, error)
+
+	// Writes metadata for a particular chunk, after each chunkserver has received a preparation message for this write.
+	// Only performs the write if the version matches.
+	CommitWrite(chunk ChunkNum, version Version, hash CommitHash) (Version, error)
+
+	// Allocates a new chunk, all zeroed out. The version number will be zero, so the only way to access it initially is
+	// with a version of AnyVersion.
+	// If this chunk isn't written to before the connection to the server closes, the empty chunk will be deleted.
+	New(aci AccessControlID) (ChunkNum, error)
+
+	// Destroys an old chunk, assuming that the metadata version matches. This includes sending messages to all relevant
+	// chunkservers.
+	Delete(num ChunkNum, version Version)
```

Comment:
Why num instead of chunk or ref, like in other places?

Reply:
No good reason.